src/gui/fspathedit_p.cpp

```diff
@@ -148,13 +154,16 @@ QValidator::State Private::FileSystemPathValidator::validate(const QList<QString
     for (int i = firstComponentToTest; i <= lastComponentToTest; ++i)
     {
         const bool isFinalPath = (i == (pathComponents.size() - 1));
-        const QStringView componentPath = pathComponents[i];
-        if (componentPath.isEmpty()) continue;
 
-        m_lastTestResult = testPath(Path(pathComponents[i].toString()), isFinalPath);
+        // construct full path name for the path tester
+        const QStringView fullComponentPath = QStringView(pathComponents.at(0).begin(), pathComponents.at(i).end());
```

Comment:
This line looks incorrect. What do you really want to do?

Reply:
This will create a string view to the full path of the i-th path component.

Example list of path components:
[ /, Level_1, Level_2, ... , Level_i, ... , Level_n ]
the i-th full component path will look like this: 
/Level_1/Level_2/.../Level_i

Reply:
But it is generally incorrect to set some range using iterator from different iterable objects even if by some lucky chance it works in some case.

Reply:
True. Redone by building a list of string views of full path components instead:
`[ /, /Level_1, /Level1/Level_2, ... ,/Level1/Level2/.../Level_n ]`